server/eventlog.go

```diff
@@ -119,12 +119,18 @@ func getVerifiedCosState(coscel cel.CEL, pcrs *tpmpb.PCRs) (*pb.AttestedCosState
 	cosState.Container.EnvVars = make(map[string]string)
 	cosState.Container.OverriddenEnvVars = make(map[string]string)
 
+	seenSeparator := false
 	for _, record := range coscel.Records {
-		// ignore non COS CEL events
-		if !record.Content.IsCosTlv() {
+		// COS State only comes from the CosEventPCR
+		if record.PCR != cel.CosEventPCR {
 			continue
 		}
 
+		// The Content.Type is not verified at this point, so we have to fail
+		// if we see any events that we do not understand. This ensures that
+		// we either verify the digest of event event in this PCR, or we fail
+		// to replay the event log.
+		// TODO: See if we can fix this to have the Content Type be verified.
```

Comment:
I believe the Content Type is verified below in `cel.VerifyDigests`. That marshals the TLV (including type) and writes the hash. It then compares the digest to digests in the CEL digest map.

Reply:
If it's an unknown Content Type as well (but verified), we should throw an error

Reply:
So the issue is that the current implementation can only call `cel.VerifyDigests` can only be called _after_ `record.Content.ParseToCosTlv()` which checks the `content.Type` field without verifying it.

So if we run into unknown Content Types, they will be unverified. This isn't a problem in practice, as we will fail on unknown content types, and will fail on known content types that fail to verify.

Reply:
Ah I see. This makes sense. I think we don't even have to call `VerifyDigests`. We can just call `record.Content.MarshalBinary()` and check the hashed value against the CEL digest map. After that, the Type is verifier and the TLV can be parsed to a CosTLV